File: pyTools/sorting_algorithms.py
def bubble_Sort(array):
    change = True
    offset = 0

    while change == True:
        change = False
        offset += 1
        
        for i in range(0, len(array) - offset):
            bubbleLeft = array[i]

            if bubbleLeft > array[i+1]:
                array[i] = array[i+1]
                array[i+1] = bubbleLeft
                change = True
    
    return array

def insertion_Sort(array):
    for i in range(1, len(array)):
        pointer = i

        while pointer > 0 and array[pointer] < array[pointer - 1]:
            buffer = array[pointer]
            array[pointer] = array[pointer - 1]
            array[pointer - 1] = buffer
            pointer -= 1
    
    return array

def merge_Sort(array):
    mainArray = []
    auxArray = []
    subArray = []

    for item in array:
        mainArray.append([item])
    
    while len(mainArray) != 1:
        for i in range(0, len(mainArray) - 1, 2):
            subcounter1 = 0
            subcounter2 = 0

            for t in range(0, len(mainArray[0]) * 2):

                if subcounter1 >= len(mainArray[i]):
                    for z in range(subcounter2, len(mainArray[i+1])):
                        subArray.append(mainArray[i+1][z])
                    break

                elif subcounter2 >= len(mainArray[i+1]):
                    for z in range(subcounter1, len(mainArray[i])):
                        subArray.append(mainArray[i][z])
                    break
              
                if mainArray[i][subcounter1] < mainArray[i+1][subcounter2]:
                    subArray.append(mainArray[i][subcounter1])
                    subcounter1 += 1
                else:
                    subArray.append(mainArray[i+1][subcounter2])
                    subcounter2 += 1
            
            auxArray.append(subArray)
            subArray = []
                
        if len(mainArray) % 2:
            auxArray.append(mainArray[len(mainArray) - 1])
    
        mainArray = auxArray
        auxArray = []
    
    return mainArray[0]

File: jsTools/sortingAlgorithms.js
function bubbleSort(array) {
    let swap = true;
    let offset = 0;
    let left;

    while (swap) {
        swap = false;
        offset++;

        for (let i = 0; i < array.length - offset; i++) {
            left = array[i];

            if (left > array[i+1]) {
                array[i] = array[i+1];
                array[i+1] = left;
                swap = true;
            }
        }
    }

    return array;
}

function insertionSort(array) {
    let pointer;
    let buffer;

    for (let i = 1; i < array.length; i++) {
        pointer = i;

        while (pointer > 0 && array[pointer] < array[pointer - 1]) {
            buffer = array[pointer];
            array[pointer] = array[pointer - 1];
            array[pointer - 1] = buffer;
            pointer--;
        }
    }

    return array;
}

let arr = [124, 15, 7241, 712, 3, 88, 12, 789, 123214, 5];
let arr2 = insertionSort(arr);

for (let i = 0; i < arr2.length; i++) {
    console.log(arr2[i]);
}